### PR TITLE
Switch to AdoptOpenJDK15 from OpenJDK15

### PR DESCRIPTION
**Pull request by andsel**
## What does this PR do?
Invalidate commit 8c3ff551ede17e23102b6857bb8cb233969ccfd9 of of PR #12537 , moves from OpenJDK 15 to AdoptOpenJDK 15

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files (and/or docker env variables)
- [ ] I have added tests that prove my fix is effective or that my feature works

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

**Comment by andsel** [comment]
Jenkins test this please

File: .ci/matrix-runtime-javas.yml [hunk 1]
```diff
@@ -7,8 +7,8 @@
 LS_RUNTIME_JAVA:
   - java8
   - openjdk11
-  - openjdk15
+  - openjdk14
   - adoptopenjdk11
-  - adoptopenjdk14
+  - adoptopenjdk15
```
Comment (roaksoax) [inline .ci/matrix-runtime-javas.yml:12]: shouldn't you continue to have adoptopenjdk14 & 15?

**Review by roaksoax — CHANGES_REQUESTED** [review]

**Review by robbavey — COMMENTED** [review]
If this is our list going forward and we need to claim support for `jdk15`, we should probably have more tests on `jdk15` than `jdk14` - maybe keep `openjdk15`, add `adoptopenjdk15`, and replace `zulu14` with `zulu15`?

I'd also like to see a `jdk16` option to see where we are before that goes GA next month.

**Comment by andsel** [comment]
Would be ideal to have the last LTS and last released versions of JDK. OpenJDK 15 has problems with centos6 while [AdoptOpenJDK 15.0.1 seems to work](https://github.com/AdoptOpenJDK/openjdk-build/issues/2097) and the [switch back of Elasticsearch](https://github.com/elastic/elasticsearch/issues/64026)  back to Adopt from Open is base don that error

**Comment by robbavey** [comment]
I agree - we definitely want broadest coverage on LTS versions of Java, the most recent regular release, and, ideally, the ability to test against the next upcoming release, to give us more insight into potential breakages coming up to the release data

**Review by roaksoax — APPROVED** [review]

**Comment by andsel** [comment]
I've added also Zulu15 in the mix so for JDK15 we keep out only OpenJDK15

**Review by robbavey — APPROVED** [review]

**Pull request merged**